pixtools/clusters/noise_analysis_SD_kmeans_Clustering.py

```diff
@@ -0,0 +1,178 @@
+# Unlike other noiseplot file (by channel) this file will cluster the SDs, allowing for a mean square analysis
+# If there are distinct clusters able to be seperated by depth, it will indicate that there is a clear relationship to noise
+# TODO: Test Functionality after noise analysis is complete, push to pixtools.
+
+# First import required packages
+import sys
+import json
+
+sys.path.insert(
+    0, "/home/s1735718/PixelsAnalysis/pixels"
+)  # Line will allow me to debug base.py code locally rather than github copy
+sys.path.append("/home/s1735718/PixelsAnalysis/pixels-analysis")
+
+from turtle import fd
+from base import *
+from channeldepth import *
+from channeldepth import *
+from sklearn.cluster import KMeans
+
+from pixels import Experiment
+from pixels.behaviours.leverpush import LeverPush
+from pixels.behaviours.pushpull import PushPull
+from pixels.behaviours.reach import Reach
+from pixels.behaviours.no_behaviour import NoBehaviour
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import datetime
+import matplotlib.pyplot as plt
+
+from pixtools import clusters
+from pixtools import utils
+
+# Now we shall read in the experimental data, defined below
+# Remember, this selection is defined as exps, NOT myexp which is defined in the normal base file. Check this before continuing!
+
+# First define the experiment to analyse and sessions
+noise_tests = Experiment(
+    ["noisetest1"],
+    NoBehaviour,
+    "~/duguidlab/visuomotor_control/neuropixels",
+)
+
+noise_test_unchanged = Experiment(
+    ["noisetest_unchanged"],
+    NoBehaviour,
+    "~/duguidlab/visuomotor_control/neuropixels",
+)
+
+noise_test_nopi = Experiment(
+    ["noisetest_nopi"],
+    NoBehaviour,
+    "~/duguidlab/visuomotor_control/neuropixels",
+)
+
+noise_test_no_caps = Experiment(
+    "VR50",
+    Reach,
+    "~/duguidlab/visuomotor_control/neuropixels",
+)
+
+# This contains the list of experiments we want to plot noise for, here only interested in the reaching task
+# Remember to change this in base.py
```

Comment:
`pixtools` should be general functions that anyone could use - this code here is not general, it's using specific experiments for something

Reply:
Likewise the `sys.path` stuff at the top, which is incidental to your setup